roles/ec2_instance_create_delete/meta/argument_specs.yml

```diff
@@ -0,0 +1,114 @@
+---
+argument_specs:
+  main:
+    short_description: A role to create an EC2 instance with optional networking resources.
+    description:
+      - A role to create an EC2 instance.
+      - Optionally can create a security group and associate an Elastic IP with the instance.
+      - Supports custom configurations for instance settings, including instance type, AMI, key pair, tags, VPC/subnet, and networking configurations.
+    options:
+      ec2_instance_create_delete_operation:
+        description:
+          - Whether to create or delete resources using the role.
+        required: false
+        type: str
+        default: create
+        choices: [create, delete]
+      ec2_instance_create_delete_aws_region:
+        description:
+          - The AWS region in which to create the EC2 instance.
+        required: true
+        type: str
+      ec2_instance_create_delete_instance_name:
+        description:
+          - The name of the EC2 instance to be created.
+        required: true
+        type: str
+      ec2_instance_create_delete_instance_type:
+        description:
+          - The instance type for the EC2 instance.
+        required: true
+        type: str
+      ec2_instance_create_delete_ami_id:
```

Comment:
If these are required here then we can't delete the instance without providing them, which I don't think should be necessary.